pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -188,97 +200,120 @@ func (cmvt *cdcMixedVersionTester) waitForResolvedTimestamps() versionStep {
 
 			cmvt.timestampsResolved.C = nil
 		}()
+		return nil
 	}
 }
 
-// finishTest marks the test as finished which will then prompt the
-// changefeed validator loop to return. This step will block until the
-// validator has finished; this is done to avoid the possibility of
-// the test closing the database connection while the validator is
-// running a query.
-func (cmvt *cdcMixedVersionTester) finishTest() versionStep {
-	return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
-		t.L().Printf("waiting for background tasks to finish")
-		cmvt.testFinished = true
-		<-cmvt.validatorFinished
+// runKafkaConsumer creates a CDC validator to validate that a changefeed
+// created on the target table is able to re-create the table
+// somewhere else. It can also verify changefeed ordering guarantees.
+func (cmvt *cdcMixedVersionTester) setupValidator() mixedversion.UserFunc {
+	return func(ctx context.Context, l *logger.Logger, r *rand.Rand, h *mixedversion.Helper) error {
+		tableName := fmt.Sprintf("%s.%s", targetDB, targetTable)
+		_, db := h.RandomDB(r, cmvt.crdbNodes)
+		if _, err := db.Exec(
+			"CREATE TABLE fprint (id INT PRIMARY KEY, balance INT, payload STRING)",
+		); err != nil {
+			return err
+		}
+
+		// NB: The fingerprint validator will save this db connection and use it
+		// when we submit rows for validation.
+		fprintV, err := cdctest.NewFingerprintValidator(db, tableName, `fprint`, cmvt.kafka.consumer.partitions, 0)
```

Comment:
The previous test would connect to random nodes when it needed a database connection. I believe we should continue to do that here -- i.e., call `DBFunc()` on the validator.